File: domain/notice.go
package domain

import "time"

type Notice struct {
	ID             string
	OrganizationID string
	Title          string
	Content        string
	Tags           []string
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

type NoticeFilter struct {
	// filter
	OrganizationID string
	Tags           []string

	// sort
	SortBy    string
	SortOrder SortOrder

	// Pagination
	Page  int64
	Limit int64
}

File: types/notice.ts
export interface Procedure {
  id: string;
  name: string;
}

export default interface Notice {
  id: string;                    // maps to id
  organization_id: string;       // maps to organization_id
  title: string;                 // maps to title
  content: string;               // maps to content
  tags: string[];                // maps to tags array
  created_at: string;            // maps to created_at
  updated_at: string;            // maps to updated_at
}
